tests/conftest.py

```diff
@@ -205,7 +205,19 @@ def pytest_configure():
         from pybind11_tests.eigen import have_eigen
     except ImportError:
         have_eigen = False
-    pypy = platform.python_implementation() == "PyPy"
+
+    # Provide simple `six`-like aliases.
+    pytest.PY2 = (sys.version_info.major == 2)
+    pytest.PY3 = (sys.version_info.major == 3)
+    pytest.CPYTHON = (platform.python_implementation() == "CPython")
+    pytest.PYPY = (platform.python_implementation() == "PyPy")
+
+    if not pytest.PY2 and not pytest.PY3:
+        print("pybind11 is only supported on Python 2.x or 3.x")
```

Comment:
Maybe just check `sys.version_info.major >= 2`? I'm not really sure how necessary this check is.

Reply:
This might indeed break if they somehow decide Python version numbers are decimal and will jump from 3.9 to 4.0?

Reply:
3.10 is next, but 4.0 won't be such a large change as 3.0.

Please remove this "fail fast" check - if someone tries Python 4.0. they should see what is going wrong so they can fix it, not have the tests fail to run. I think you should remove the CPython/PyPy check too - if someone tries to run the test, they should see what goes wrong. It should fail to compile, generally, so they shouldn't get here.

I generally don't like fail-fast checks (I spent hours trying to debug an error that was hidden behind a fail-fast check last week), and you should never fail-fast for a version of something that doesn't exist yet.

Reply:
I'm OK with removing the fail-fast, but I dunno how I feel about there being no warning at all. Maybe I'll try to add a warning in after this merges.

I guess I can feel your pain about the fail-fast, but I do think fail-fasts can help in certain contexts.

Reply:
If it fails, the warning will be the actual failure. And if it works, hooray!

The one I almost didn't remove was the CPython + PyPy one, but I have no idea how this would even reach the Python stage, it shouldn't compile.

Reply:
Er, yeah, but failing fast in contract-checking can help, e.g.
"Hey, I'm a function that will gain absolutely zero from duck typing, and I want a specific type... but I dont' have `assert isinstance(...)`. Now *you* get to figure out why this random field or dictionary string is missing!"

So yeah, that's what I mean about "in certain contexts". For build systems, warnings are perhaps best, but some users completely ignore warnings too...